com.ibm.streamsx.topology/opt/python/packages/streamsx/rest_primitives.py

```diff
@@ -1182,6 +1182,45 @@ def __init__(self, rest_client, credentials):
     def _get_url(self, req_name):
         return self._credentials['rest_url'] + self._credentials[req_name]
 
+    def submit_job(self, sab_file, application_name,  parameters=[], configuration={}, tracing='info' ):
+        """Submit a compiled Streams Application Bundle (a SAB file) to run in this Streamin Analytics service.
+        
+        Args:
+        sab_file (string): path to the compiled SAB file containing the application to be submitted
+        application_name (string): name of the application in the SAB file
+        parameters (list of {name/value} pairs, optional): application's submission-time parameters
+        configuration (dict, optional): hash of deployment configuration parameters 
```

Comment:
I think we need to design the api to support job config overlays fully rather than just a subset.
We probably have at least these cases:

- Pass in a complete JCO file/JSON - allows full flexibility.
- Pass in a JobConfig object which is the Python binding to a JCO.

Setting individual items might be better handled through the JobConfig object instead of a dict of key value pairs.

Reply:
Agreed -- I've moved the 'target_pe_count' and 'submission_parameters' arguments to the JobConfig constructor, and changed the submit_job() method to optionally accept a job configuration overlay.